code/modules/events/carp_migration.dm

```diff
@@ -6,6 +6,14 @@
 	earliest_start = 10 MINUTES
 	max_occurrences = 6
 
+/datum/round_event_control/carp_migration/New()
+	. = ..()
+	if(HAS_TRAIT(SSstation, STATION_TRAIT_CARP_INFESTATION))
```

Comment:
Early return

Reply:
man its one layer deep the early return will literaly not make a difference :(